source/extensions/transport_sockets/proxy_protocol/proxy_protocol.h

```diff
@@ -37,6 +37,8 @@ class UpstreamProxyProtocolSocket : public TransportSockets::PassthroughSocket,
   Network::TransportSocketCallbacks* callbacks_{};
   Buffer::OwnedImpl header_buffer_{};
   ProxyProtocolConfig_Version version_{ProxyProtocolConfig_Version::ProxyProtocolConfig_Version_V1};
+  bool pass_all_tlvs_ = false;
```

Comment:
If these values are immutable, please make them `const`.

Reply:
It is mutable at beginning based on the config match type. It is related to API setting.. cc @envoyproxy/api-shepherds as well.

Reply:
We can move to the method defination like, 
```
pass_all_tlvs_(proto_config.has_pass_through_tlvs() ? proto_config.pass_through_tlvs().match_type() == ProxyProtocolPassThroughTLVs::INCLUDE_ALL
                     : false)
```
But the we still need to check the `INCLUDE` type, and looks like the current is cleaner, WDYT?